airbyte-integrations/connectors/source-snowflake/source_snowflake/streams.py

```diff
@@ -0,0 +1,454 @@
+                                                     **stream_filtered_kwargs)
+        self._namespace = None
+
+    @property
+    def namespace(self):
+        return self._namespace
+
+    @namespace.setter
+    def namespace(self, namespace):
+        self._namespace = namespace
+
+    @property
+    def name(self):
+        return f"{self.table_object['schema']}.{self.table_object['table']}"
+
+    def path(
+            self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        """
+            path of request
+        """
+
+        return f"{self.url_base}/{self.url_suffix}"
+
+    @property
+    def url_base(self):
+        return self._url_base
+
+    @property
+    def statement(self):
+        database = self.config["database"]
+        schema = self.table_object["schema"]
+        table = self.table_object["table"]
+
+        return f"SELECT * FROM {database}.{schema}.{table}"
+
+    def request_body_json(
+            self,
+            stream_state: Optional[Mapping[str, Any]],
+            stream_slice: Optional[Mapping[str, Any]] = None,
+            next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Optional[Mapping[str, Any]]:
+        json_payload = {
+            "statement": self.statement,
+            "role": self.config['role'],
+            "warehouse": self.config['warehouse'],
+            "database": self.config['database'],
+            "timeout": "1000",
+        }
+
+        schema = self.table_object.get('schema', '')
+        if schema:
+            json_payload['schema'] = schema
+
+        return json_payload
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        response_json = response.json()
+        ordered_column_names = [row_type['name'] for row_type in
+                                response_json.get('resultSetMetaData', {'rowType': []}).get('rowType', [])]
+        for record in response_json.get("data", []):
+            yield {column_name: column_value for column_name, column_value in zip(ordered_column_names, record)}
+
+    def __str__(self):
+        return f"Current stream has this table object as constructor {self.table_object}"
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        properties = {}
+        json_schema = {
+            "$schema": "https://json-schema.org/draft-07/schema#",
+            "type": "object",
+            "additionalProperties": True,
+            "properties": properties,
+        }
+
+        for column_object in self.table_schema_stream.read_records(sync_mode=SyncMode.full_refresh):
+            column_name = column_object['column_name']
+            snowflake_column_type = column_object['type'].lower()
+            airbyte_column_type_object = mapping_snowflake_type_airbyte_type.get(snowflake_column_type, )
```

Comment:
This seems unsafe as if there is a type we don't know about, we will have an invalid schema

Reply:
I asked for the default value of a type. This was your answer: set unknown type to be a string ([example](https://github.com/airbytehq/airbyte/blob/e88c0b120d91ec053deecde6f3d4ec7e75d292d1/airbyte-integrations/connectors/source-airtable/source_airtable/schema_helpers.py#L125-L127))
Benefits: Streams are always available given a type we don’t know
Drawback: when updating the code to support the new type, it will probably be a breaking change as the type will change from “string” to the new type
raise an error so that we fix the issue ([example](https://github.com/airbytehq/airbyte/blob/e88c0b120d91ec053deecde6f3d4ec7e75d292d1/airbyte-integrations/connectors/source-salesforce/source_salesforce/api.py#L453-L454))
Benefits: Ensure data integrity
Drawback: Stream is not available when a unknown type is added and a new release is required.

You said also to go with option 2, I will create a detailed error to give more details about the nature of the error.

Reply:
I'm still fine with option. I'm glad you added the error. Can we make sure the error is properly actionable for the customer? In this case, I think the action would be to contact Airbyte support to update the connector to handle those new fields

Reply:
The error message is updated now.